x/observer/keeper/msg_server_disable_fast_confirmation_test.go

```diff
@@ -0,0 +1,199 @@
+		msg := types.MsgDisableFastConfirmation{Creator: admin, ChainId: chainParams.ChainId}
+		keepertest.MockCheckAuthorization(&authorityMock.Mock, &msg, authoritytypes.ErrUnauthorized)
+		_, err := srv.DisableFastConfirmation(ctx, &msg)
+
+		// ASSERT
+		require.ErrorIs(t, err, authoritytypes.ErrUnauthorized)
+
+		// chain params list should be unchanged
+		chainParamsList, found := k.GetChainParamsList(ctx)
+		require.True(t, found)
+		require.Len(t, chainParamsList.ChainParams, 1)
+		require.Equal(t, chainParams, chainParamsList.ChainParams[0])
+	})
+
+	t.Run("cannot disable fast confirmation if chain params not found", func(t *testing.T) {
+		// ARRANGE
+		k, ctx, _, _ := keepertest.ObserverKeeperWithMocks(t, keepertest.ObserverMockOptions{
+			UseAuthorityMock: true,
+		})
+
+		// create mock admin, msg server and authority keeper
+		admin := sample.AccAddress()
+		srv := keeper.NewMsgServerImpl(*k)
+		authorityMock := keepertest.GetObserverAuthorityMock(t, k)
+
+		// ACT
+		msg := types.MsgDisableFastConfirmation{Creator: admin, ChainId: 1}
+		keepertest.MockCheckAuthorization(&authorityMock.Mock, &msg, nil)
+		_, err := srv.DisableFastConfirmation(ctx, &msg)
+
+		// ASSERT
+		require.ErrorIs(t, err, types.ErrChainParamsNotFound)
+	})
+
+	t.Run("cannot disable fast confirmation if validation fails", func(t *testing.T) {
+		// ARRANGE
+		k, ctx, _, _ := keepertest.ObserverKeeperWithMocks(t, keepertest.ObserverMockOptions{
+			UseAuthorityMock: true,
+		})
+
+		// create mock admin, msg server and authority keeper
+		admin := sample.AccAddress()
+		srv := keeper.NewMsgServerImpl(*k)
+		authorityMock := keepertest.GetObserverAuthorityMock(t, k)
+
+		// create invalid chain params
+		chainList := chains.ExternalChainList([]chains.Chain{})
+		chainParams := sample.ChainParams(chainList[0].ChainId)
+		chainParams.ConfirmationParams.SafeInboundCount = 0 // <- invalid count
+
+		// setup chain params list
+		setupChainParamsList(t, k, ctx, admin, []*types.ChainParams{chainParams})
+
+		// ACT
+		// try to disable FAST confirmation for the chain
+		msg := types.MsgDisableFastConfirmation{Creator: admin, ChainId: chainParams.ChainId}
+		keepertest.MockCheckAuthorization(&authorityMock.Mock, &msg, nil)
+		_, err := srv.DisableFastConfirmation(ctx, &msg)
+
+		// ASSERT
+		require.ErrorIs(t, err, types.ErrInvalidChainParams)
+
+		// chain params list should be unchanged
+		chainParamsList, found := k.GetChainParamsList(ctx)
+		require.True(t, found)
+		require.Len(t, chainParamsList.ChainParams, 1)
+		require.Equal(t, chainParams, chainParamsList.ChainParams[0])
+	})
+}
+
+// ensureFastConfirmationEnabled checks that the fast confirmation is enabled
+func ensureFastConfirmationEnabled(t *testing.T, chainParams *types.ChainParams) {
+	require.NotNil(t, chainParams.ConfirmationParams)
+	require.Positive(t, chainParams.ConfirmationParams.FastInboundCount)
+	require.Positive(t, chainParams.ConfirmationParams.FastOutboundCount)
```

Comment:
I think it's not up to date, should check lower than safe

Reply:
adjusted sample function https://github.com/zeta-chain/node/pull/3522/commits/05ece1e475ab004869bd8297f35127d5d9870f31